compress.cc

```diff
@@ -81,8 +81,8 @@ compressor::ptr_type compressor::create(const sstring& name, const opt_getter& o
     return compressor_registry::create(qn, opts);
 }
 
-shared_ptr<compressor> compressor::create(const std::map<sstring, sstring>& options) {
-    auto i = options.find(compression_parameters::SSTABLE_COMPRESSION);
+shared_ptr<compressor> compressor::create(const std::map<sstring, sstring>& options, opt_string class_opt) {
+    auto i = options.find(class_opt.value_or(compression_parameters::SSTABLE_COMPRESSION));
```

Comment:
I think it's nicer if this function receives a struct for parameters and leaves the string_map (which I now understand, but which should be documented fully in config) parsing to its caller.

Reply:
But, we can leave that for the indefinite future.

Reply:
But please make class_opt a required parameter and update the callers, to reduce magic. SSTABLE_COMPRESSION is not some kind of privileged or reasonable default value.

Reply:
Oh, I agree. I just don't want to contaminate to much sstable code in this PR...

Reply:
It's okay. Another option is to separate it out.